src/io/unix/iouring.rs

```diff
@@ -69,31 +69,35 @@ impl File {
         self.sync.size()
     }
 
-    pub(crate) async fn write_append_all(&self, buf: Bytes) -> IOResult<()> {
+    pub(crate) async fn write_append_writable_data<R: Send + 'static>(
+        &self,
+        c: impl WritableDataCreator<R>,
+        len: u64,
```

Comment:
Do not pass `len` as argument. This is not safe. You should calculate `len` within this function